packages/mui-material/src/StepLabel/StepLabel.js

```diff
@@ -154,7 +156,19 @@ const StepLabel = React.forwardRef(function StepLabel(inProps, ref) {
 
   const classes = useUtilityClasses(ownerState);
 
-  const labelSlotProps = slotProps.label ?? componentsProps.label;
+  const externalForwardedProps = {
+    slots,
+    slotProps: {
+      ...componentsProps,
+      ...slotProps,
+    },
+  };
+
+  const [StepLabelLabelSlot, stepLabelLabelProps] = useSlot('label', {
```

Comment:
nit:

```suggestion
  const [LabelSlot, labelProps] = useSlot('label', {
```

Can we do that? I think it's simpler